### PR TITLE
No dataset module changed

### LINKED ISSUE #2970: Fix layout problem with no-datasets blue element
screen shot 2015-03-31 at 11 04 01

### LINKED ISSUE #4292: Add a visual tip to indicate the search box for when not having datasets and being prompted to the 'data library list'

### PR DESCRIPTION
**Pull request by xavijam**
- [x] Fixed problem with navigation menu when no-datasets module is present in (Fixes #2970) cc @piensaenpixel. 
- [x] No datasets module reviewed in library section within new dashboard (Fixes #4292).
- [x] No datasets module reviewed in library section within create dialog (Fixes #4292).
- [x] Reviewed navigation bar for add-layer-view in editor page.

---

screen shot 2015-08-25 at 12 46 08

---

screen shot 2015-08-25 at 12 46 16

---

REVIEWER: @javierarce 
UI: @josecruz 
cc @viddo

### CODE REVIEW

**Comment by xavijam** [comment]
retest this please

**Comment by xavijam** [comment]
retest this please

File: lib/assets/javascripts/cartodb/common/views/create/listing/content_no_datasets.jst.ejs [hunk 1]
```diff
@@ -1,2 +1,3 @@
 <h4 class="IntermediateInfo-title">You have not connected any datasets yet</h4>
-<p class="DefaultParagraph">Start with one of the following datasets from the data library.</p>
+<p class="DefaultParagraph">You can <button class="Button--link js-connect">Connect your own dataset</button> or choose one from our Data Library. You can <strong>drag and drop</strong> your dataset as well.</p>
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/common/views/create/listing/content_no_datasets.jst.ejs]: Here "Connect" appears in capitals. In the other template (`content_no_datasets.jst.ejs`) appears in downcase. I'll go for the downcase option.

File: lib/assets/javascripts/cartodb/dashboard/views/content_no_datasets.jst.ejs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 <h4 class="IntermediateInfo-title">You have not connected any datasets yet</h4>
 <p class="DefaultParagraph">
-  <button class="ContentResult-textTitle js-connect">Connect your data</button> or start with one of the following datasets from the data library.
+  You can <button class="ContentResult-textTitle js-connect">connect datasets</button> or <strong>Search</strong> in all our data library .
```
Comment (javierarce) [inline lib/assets/javascripts/cartodb/dashboard/views/content_no_datasets.jst.ejs]: I'd write "search" instead of "Search".

**Comment by javierarce** [comment]
:+1: code looks good to me, I just added a couple of minor comments. Also, are we testing that the view for the content message appears?

**Comment by josecruz** [comment]
One thing: why on the first screen "connect datasets" is underlined?

**Comment by xavijam** [comment]
It was underlined before...

**Comment by josecruz** [comment]
Okay so can you please remove it? So we keep the consistency on both modals

**Comment by xavijam** [comment]
:+1:

**Comment by josecruz** [comment]
Thanks!! :+1:

**Comment by xavijam** [comment]
@javierarce: https://github.com/CartoDB/cartodb/blob/no-dataset-module/lib/assets/test/spec/cartodb/dashboard/content_controller_view.spec.js#L157

**Comment by xavijam** [comment]
Tested in ded06, if you want to.

**Pull request merged**